File: app/src/main/java/com/mihaiapps/securevault/bl/glide/LocalEncryptedImageModelLoaderFactory.java
package com.mihaiapps.securevault.bl.glide;

import com.bumptech.glide.load.model.ModelLoader;
import com.bumptech.glide.load.model.ModelLoaderFactory;
import com.bumptech.glide.load.model.MultiModelLoaderFactory;

import java.io.InputStream;

import androidx.annotation.NonNull;

public class LocalEncryptedImageModelLoaderFactory implements ModelLoaderFactory<String, InputStream> {
    @NonNull
    @Override
    public ModelLoader<String, InputStream> build(@NonNull MultiModelLoaderFactory multiFactory) {
        return new LocalEncryptedImageModelLoader(new LocalEncryptedDataFetcherFactory());
    }

    @Override
    public void teardown() {

    }
}

File: app/src/main/java/com/mihaiapps/securevault/bl/Glide/LocalEncryptedImageModelLoaderFactory.kt
package com.mihaiapps.securevault.bl.Glide

import android.content.Context
import com.bumptech.glide.load.model.ModelLoader
import com.bumptech.glide.load.model.ModelLoaderFactory
import com.bumptech.glide.load.model.MultiModelLoaderFactory
import com.mihaiapps.securevault.bl.LocalFileReader
import com.mihaiapps.securevault.bl.enc.cipher.BaseCipher
import java.io.InputStream

class LocalEncryptedImageModelLoaderFactory(val context: Context, private val aesCipher: BaseCipher) : ModelLoaderFactory<String, InputStream> {
    override fun build(multiFactory: MultiModelLoaderFactory): ModelLoader<String, InputStream> {
        return LocalEncryptedImageModelLoader(
                LocalEncryptedDataFetcherFactory(LocalFileReader(context),aesCipher))
    }

    override fun teardown() {
    }

}